sys/mips/gt64120.c

```diff
@@ -335,9 +340,12 @@ static int gt_pci_attach(device_t *pcib) {
   return bus_generic_probe(pcib);
 }
 
-static bool gt_pci_bar(device_t *dev, int rid) {
+static bool gt_pci_bar(device_t *dev, res_type_t type, int rid,
+                       rman_addr_t start) {
   pci_device_t *pcid = pci_device_of(dev);
-  return rid < PCI_BAR_MAX && pcid->bar[rid].size != 0;
+  if ((type == RT_IOPORTS && start > IO_ISAEND) || type == RT_MEMORY)
```

Comment:
Please rewrite `gt_pci_bar`, so that early exits return `false` and you move the bulk of computations to the end of function. Early exits are important to filter out cases we don't want to handle.

Reply:
Done.